mono/metadata/marshal.c

```diff
@@ -1178,6 +1178,12 @@ mono_type_to_stind (MonoType *type)
 MonoAsyncResult *
 mono_delegate_begin_invoke (MonoDelegate *delegate, gpointer *params)
 {
+#ifdef ENABLE_NETCORE
+	MonoException *exception;
+	exception = mono_exception_from_name (mono_defaults.corlib, "System", "PlatformNotSupportedException");
+	mono_set_pending_exception (exception);
+	return NULL;
+#else
```

Comment:
For reference, .NET Core doesn't support BeginInvoke/EndInvoke on delegates.

https://github.com/dotnet/corert/blob/635cf21aca11265ded9d78d216424bd609c052f5/src/Common/src/TypeSystem/IL/Stubs/DelegateMethodILEmitter.cs#L21-L26
https://github.com/dotnet/coreclr/blob/030a3ea9b8dbeae89c90d34441d4d9a1cf4a7de6/tests/src/baseservices/threading/delegate/BeginInvokeEndInvoke.cs#L29-L32

Reply:
https://github.com/dotnet/coreclr/blob/020a8c44cc9606bf06b107597cd853533981bbd3/src/vm/comdelegate.cpp#L2081-L2088

Reply:
Is this the best place to "block" this functionality?

Reply:
I will check. I wanted to submit it as separate PR but this progressed a bit faster than I expected so I didn't get to revisit it yet.

Reply:
Alternative is to emit the exception throwing directly in `emit_delegate_[begin/end]_invoke_ilgen`. I don't think it's any better.